platform/darwin/src/MGLCluster.h

```diff
@@ -0,0 +1,54 @@
+#import <Foundation/Foundation.h>
+#import "MGLFoundation.h"
+
+@protocol MGLFeature;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ An `NSUInteger` constant used to indicate an invalid cluster identifier.
+ This indicates a missing cluster feature.
+ */
+FOUNDATION_EXTERN MGL_EXPORT const NSUInteger MGLClusterIdentifierInvalid;
+
+/**
+ A protocol that (private) feature subclasses (i.e. those already conforming to
+ the `MGLFeature` protocol) conform to if they are represent clusters.
+ 
+ Currently only subclasses of `MGLPointFeature` support `MGLCluster`.
+ 
+ To check if a feature is a cluster, check conformity to `MGLCluster`, for
+ example:
+ 
+ ```swift
+ let shape = try! MGLShape(data: clusterShapeData, encoding: String.Encoding.utf8.rawValue)
+ 
+ guard let pointFeature = shape as? MGLPointFeature else {
+     throw ExampleError.unexpectedFeatureType
+ }
+ 
+ // Check for cluster conformance
+ guard let cluster = pointFeature as? MGLCluster else {
+     throw ExampleError.featureIsNotACluster
+ }
+ 
+ print("Approximate number of points in cluster: \(cluster.clusterPointCountAbbreviation)")
+ ```
+ */
+MGL_EXPORT
+@protocol MGLCluster <NSObject>
```

Comment:
Unless we expect non-features to be clusterable in the future, let’s make this protocol conform to MGLFeature.

```suggestion
@protocol MGLCluster <MGLFeature>
```